### ISSUE TITLE
Crash while loading json from OpenStreetMap with latest Graddle Plugin

### ISSUE THREAD

**Issue by binayakumar09** [koush/ion#935]
While trying to load json from OpenStreetMap, I see a crash with gradlle plugin post 4.1.0-XXX ...

   --------- beginning of crash
E/AndroidRuntime: FATAL EXCEPTION: ion-ion
    Process: com.xxxxx.yyyyyyyy, PID: 5463
    java.lang.AssertionError
        at com.koushikdutta.async.AsyncSSLSocketWrapper.write(SourceFile:474)
        at com.koushikdutta.async.AsyncSSLSocketWrapper.handleHandshakeStatus(SourceFile:378)
        at com.koushikdutta.async.AsyncSSLSocketWrapper.handshake(SourceFile:182)
        at com.koushikdutta.async.http.AsyncSSLSocketMiddleware.tryHandshake(SourceFile:89)
        at com.koushikdutta.async.http.AsyncSSLSocketMiddleware$2.onConnectCompleted(SourceFile:106)
        at com.koushikdutta.async.http.AsyncSocketMiddleware.lambda$getSocket$3(SourceFile:217)
        at com.koushikdutta.async.http.AsyncSocketMiddleware.lambda$getSocket$3$AsyncSocketMiddleware(Unknown Source:0)
        at com.koushikdutta.async.http.-$$Lambda$AsyncSocketMiddleware$glazjOxu-iNIRY4N890Wl1ngyGo.onCompleted(Unknown Source:12)
        at com.koushikdutta.async.future.SimpleFuture.lambda$setCallback$0(SourceFile:216)
        at com.koushikdutta.async.future.-$$Lambda$SimpleFuture$h_gK5Y1XHZI1H8JTE2xOg4dXbp8.onCompleted(Unknown Source:2)
        at com.koushikdutta.async.future.SimpleFuture$FutureCallsite.loop(SourceFile:130)
        at com.koushikdutta.async.future.SimpleFuture.handleCallbackUnlocked(SourceFile:153)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SourceFile:195)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SourceFile:175)
        at com.koushikdutta.async.future.-$$Lambda$VNAuWcttAhMr3VsWCERbmWE6yR0.success(Unknown Source:2)
        at com.koushikdutta.async.future.SimpleFuture.lambda$success$4(SourceFile:287)
        at com.koushikdutta.async.future.-$$Lambda$SimpleFuture$DxIsaj7NQi76eWgOKS4iviWPLV0.onCompleted(Unknown Source:4)
        at com.koushikdutta.async.future.SimpleFuture$FutureCallsite.loop(SourceFile:130)
        at com.koushikdutta.async.future.SimpleFuture.handleCallbackUnlocked(SourceFile:153)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SourceFile:195)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SourceFile:182)
        at com.koushikdutta.async.http.-$$Lambda$cBC5du-82c5EcL72rP5wNK7Y2Y4.onConnectCompleted(Unknown Source:2)
        at com.koushikdutta.async.AsyncServer.runLoop(SourceFile:906)
        at com.koushikdutta.async.AsyncServer.run(SourceFile:727)
        at com.koushikdutta.async.AsyncServer.access$800(SourceFile:46)
        at com.koushikdutta.async.AsyncServer$8.run(SourceFile:681)

**Comment by tomeroto** [comment]
I have same issues, after update my Android Studio to 4.1.

```
10-14 20:00:29.799 5110-5151/com.test.http E/AndroidRuntime: FATAL EXCEPTION: ion-ion
    Process: com.test.http, PID: 5110
    java.lang.AssertionError
        at com.koushikdutta.async.AsyncSSLSocketWrapper.write(AsyncSSLSocketWrapper.java:492)
        at com.koushikdutta.async.AsyncSSLSocketWrapper.handleHandshakeStatus(AsyncSSLSocketWrapper.java:378)
        at com.koushikdutta.async.AsyncSSLSocketWrapper.handshake(AsyncSSLSocketWrapper.java:182)
        at com.koushikdutta.async.http.AsyncSSLSocketMiddleware.tryHandshake(AsyncSSLSocketMiddleware.java:89)
        at com.koushikdutta.async.http.AsyncSSLSocketMiddleware$2.onConnectCompleted(AsyncSSLSocketMiddleware.java:106)
        at com.koushikdutta.async.http.AsyncSocketMiddleware.lambda$getSocket$3$AsyncSocketMiddleware(AsyncSocketMiddleware.java:217)
        at com.koushikdutta.async.http.-$$Lambda$AsyncSocketMiddleware$glazjOxu-iNIRY4N890Wl1ngyGo.onCompleted(lambda)
        at com.koushikdutta.async.future.SimpleFuture.lambda$setCallback$0(SimpleFuture.java:212)
        at com.koushikdutta.async.future.-$$Lambda$SimpleFuture$h_gK5Y1XHZI1H8JTE2xOg4dXbp8.onCompleted(lambda)
        at com.koushikdutta.async.future.SimpleFuture$FutureCallsite.loop(SimpleFuture.java:130)
        at com.koushikdutta.async.future.SimpleFuture.handleCallbackUnlocked(SimpleFuture.java:153)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:191)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:174)
        at com.koushikdutta.async.future.-$$Lambda$VNAuWcttAhMr3VsWCERbmWE6yR0.success(lambda)
        at com.koushikdutta.async.future.SimpleFuture.lambda$success$4(SimpleFuture.java:279)
        at com.koushikdutta.async.future.-$$Lambda$SimpleFuture$DxIsaj7NQi76eWgOKS4iviWPLV0.onCompleted(lambda)
        at com.koushikdutta.async.future.SimpleFuture$FutureCallsite.loop(SimpleFuture.java:130)
        at com.koushikdutta.async.future.SimpleFuture.handleCallbackUnlocked(SimpleFuture.java:153)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:191)
        at com.koushikdutta.async.future.SimpleFuture.setComplete(SimpleFuture.java:178)
        at com.koushikdutta.async.http.-$$Lambda$cBC5du-82c5EcL72rP5wNK7Y2Y4.onConnectCompleted(lambda)
        at com.koushikdutta.async.AsyncServer.runLoop(AsyncServer.java:896)
        at com.koushikdutta.async.AsyncServer.run(AsyncServer.java:717)
        at com.koushikdutta.async.AsyncServer.access$800(AsyncServer.java:46)
        at com.koushikdutta.async.AsyncServer$11.run(AsyncServer.java:671)
10-14 20:00:29.984 5110-5128/com.test.http E/Surface: getSlotFromBufferLocked: unknown buffer: 0x7f9a075e20
```

**Comment by akhilakhi84** [comment]
Finally i found a soultion

**Solution 1:**
Go to: `File > Invalidate Caches/Restart` and select `Invalidate and Restart`

**Solution 2:**
Go to: `Build > Select Build Variant` and select `release` instead of `debug`

**Comment by binayakumar09** [comment]
Issue resolved with Version 3.0.1
